Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -1173,10 +1181,6 @@ def dlconfig_changed_callback(self, section, name, new_value, old_value):
     def get_share_mode(self):
         return self.handle.status().share_mode
 
-    @waitForHandleAndSynchronize(True)
-    def set_share_mode(self, share_mode):
```

Comment:
AFAIK, this method is stable enough even without decorator as long as the handle exist (and ready). In the early time I implemented this I have no issue with this.